src/lib.rs

```diff
@@ -27,11 +110,197 @@
 #[doc = include_str!("../README.md")]
 mod readme {}
 
+mod format;
+
 #[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
+mod tests;
+
+use core::fmt;
+
+/// Error type returned by the `strftime` functions.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum Error {
+    /// Provided format string is ended by an unterminated format specifier.
+    InvalidFormatString,
+    /// Formatted string is too large and could cause an out-of-memory error.
+    FormattedStringTooLarge,
+    /// Provided buffer for the [`buffered::strftime`] function is too small for the formatted string.
+    ///
+    /// This corresponds to the [`std::io::ErrorKind::WriteZero`] variant.
+    ///
+    /// [`std::io::ErrorKind::WriteZero`]: <https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.WriteZero>
+    WriteZero,
+    /// Formatting error, corresponding to [`core::fmt::Error`].
+    FmtError,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Error::InvalidFormatString => write!(f, "invalid format string"),
+            Error::FormattedStringTooLarge => write!(f, "formatted string too large"),
+            Error::WriteZero => write!(f, "failed to write the whole buffer"),
+            Error::FmtError => write!(f, "formatter error"),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+#[cfg_attr(docsrs, doc(cfg(feature = "std")))]
+impl std::error::Error for Error {}
+
+/// Common methods needed for formatting _time_.
```

Comment:
Can we add some text explaining that all of the `strftime` functions take an implementation of this trait and that it is expected that `strftime::Time` be implemented for structs that represent a time?